webapp/src/types/store.ts

```diff
@@ -0,0 +1,11 @@
+import {GlobalState as ReduxGlobalState} from 'mattermost-redux/types/store';
+
+import combinedReducers from '../reducers';
+
+export type GlobalState = ReduxGlobalState & {
+    'plugins-com.github.manland.mattermost-plugin-gitlab': PluginState
+};
+
+export type PluginState = ReturnType<typeof combinedReducers>
+
+export type pluginStateKey = 'plugins-com.github.manland.mattermost-plugin-gitlab'
```

Comment:
We can do use `as const` so we can use this value:

```suggestion
export const pluginStateKey = 'plugins-com.github.manland.mattermost-plugin-gitlab' as const;
```